cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -70,7 +70,27 @@ private URI addCookieInterceptorIfRequired(URI uri) {
         if (uri.getUserInfo() != null) {
             String[] parts = uri.getUserInfo().split(":");
             if (parts.length == 2) {
-                CookieInterceptor ci = new CookieInterceptor(parts[0], parts[1], uri.toString());
+
+                String path = uri.getRawPath();
+                int index = path.lastIndexOf("/");
+                if (index == path.length() - 1) {
+                    // we need to go back one
+                    path = path.substring(0, index);
+                    index = path.lastIndexOf("/");
+                }
+
+                path = path.substring(0, index);
+
+                URI baseURI;
+
+                try {
+                    baseURI = new URI(uriProtocol, null, uriHost, uriPort, path, null, null);
+                } catch (URISyntaxException e) {
+                    throw new RuntimeException(e);
```

Comment:
I don't like this, but as it is already the approach defined here I think it just needs to get covered under #280 for v2 API exceptions.